accounts/tests.py

```diff
@@ -669,3 +670,79 @@ def test_user_delete_remove_sounds(self):
         self.assertEqual(DeletedSound.objects.filter(user__id=user.id).exists(), True)
 
 
+class UserEmailsUniqueTestCase(TestCase):
+
+    def setUp(self):
+        self.user_a = User.objects.create_user("user_a", password="12345", email='a@b.com')
+        self.user_b = User.objects.create_user("user_b", password="12345", email='c@d.com')
+        self.user_c = User.objects.create_user("user_c", password="12345", email=generate_tmp_email('c@d.com'))
+        SameUser.objects.create(
+            main_user=self.user_b,
+            main_orig_email=self.user_b.email,
+            secondary_user=self.user_c,
+            secondary_orig_email=self.user_b.email,  # Must be same email (original)
+        )
+        # User a never had problems with email
+        # User b and c had the same email, but user_c's was automaitcally changed to avoid duplicates
+
+    def test_redirects_when_shared_emails(self):
+
+        # Try to log-in with user and go to messages page (any login_required page would work)
+        # User a is not in same users table, so redirect should be plain and simple to messages
+        resp = self.client.post(reverse('login'),
```

Comment:
Perhaps a comment here to say that we POST to login because the email check logic is done in the login view - this is why we can't use `self.client.login`